### PR TITLE
Protocol Features

### PR DESCRIPTION
**Pull request by DmitryDodzin**
Add features to protocol to allow limiting for backward/forwards compatibility

required for #1742

### CODE REVIEW

File: mirrord/agent/src/cli.rs [hunk 1]
```diff
@@ -36,6 +37,9 @@ pub struct Args {
     #[arg(short = 'p', long, default_value_t = false)]
     pub pause: bool,
 
+    #[arg(long, default_value_t = Features::all())]
+    pub protocol_features: Features,
```
Comment (aviramha) [inline mirrord/agent/src/cli.rs:41]: Why we would want users to specify that?
Also, why wouldn't it just be v1/v2?

Comment (DmitryDodzin) [reply mirrord/agent/src/cli.rs:41]: It's optional to clamp the agent to a certain version so we can clamp it from the operator if needed

Comment (aviramha) [reply mirrord/agent/src/cli.rs:41]: but why is it features and not just a version?

File: mirrord/protocol/src/features.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+impl FromStr for Features {
+    type Err = bitflags::parser::ParseError;
+
+    fn from_str(flags: &str) -> Result<Self, Self::Err> {
+        Ok(Self(flags.parse()?))
+    }
+}
+
+thread_local!(
```
Comment (aviramha) [inline mirrord/protocol/src/features.rs:28]: This is ugly, what can we do about it?
I'd prefer it to be a OnceLock/LazyCell - no need for thread local..

**Pull request closed without merging**